back/boxtribute_server/models/crud.py

```diff
@@ -65,6 +80,170 @@ def create_box(
+            product=box.product.id,
+            size=box.size.id,
+            defaults={"number_of_items": 0, "location": box.location.id},
+        )
+
+        unboxed_items_collection.number_of_items += number_of_items
+        box.items -= number_of_items
+
+        unboxed_items_collection.save()
+        box.save()
+        return unboxed_items_collection
+
+
+def move_box_to_distribution_event(box_label_identifier, distribution_event_id):
+    """Move a box to a distribution event."""
+    with db.database.atomic():
+        box = Box.get(Box.label_identifier == box_label_identifier)
+        distribution_event = DistributionEvent.get_by_id(distribution_event_id)
+        # Completed Events should not be mutable anymore
+        distribution_event = DistributionEvent.get_by_id(distribution_event_id)
+        if distribution_event.state == DistributionEventState.Completed:
+            raise ModifyCompletedDistributionEvent(
+                desired_operation="move_box_to_distribution_event",
+                distribution_event_id=distribution_event.id,
+            )
+        box.location = distribution_event.distribution_spot_id
+        box.distribution_event = distribution_event_id
+        box.save()
+        return box
+
+
+def change_distribution_event_state(distribution_event_id, distribution_event_state):
+    distribution_event = DistributionEvent.get_by_id(distribution_event_id)
+
+    # Completed Events should not be mutable anymore
+    if distribution_event.state == DistributionEventState.Completed:
+        raise InvalidDistributionEventState(
+            expected_states=[
+                s
+                for s in DistributionEventState
+                if s != DistributionEventState.Completed
+            ],
+            actual_state=distribution_event_state,
+        )
+
+    distribution_event.state = distribution_event_state
+    distribution_event.save()
+    return distribution_event
+
+
+def add_packing_list_entry_to_distribution_event(
+    user_id,
+    distribution_event_id,
+    product_id,
+    size_id,
+    number_of_items,
+):
+    """
+    Add a packing list entry to a distribution event.
+    """
+    now = utcnow()
+
+    # Completed Events should not be mutable anymore
+    distribution_event = DistributionEvent.get_by_id(distribution_event_id)
+    if distribution_event.state == DistributionEventState.Completed:
+        raise ModifyCompletedDistributionEvent(
+            desired_operation="add_packing_list_entry",
+            distribution_event_id=distribution_event_id,
+        )
+
+    existing_packing_list_entry = PackingListEntry.get_or_none(
+        PackingListEntry.distribution_event == distribution_event_id,
+        PackingListEntry.product == product_id,
+        PackingListEntry.size == size_id,
+    )
+    if existing_packing_list_entry is not None:
+        if number_of_items > 0:
+            existing_packing_list_entry.number_of_items = number_of_items
```

Comment:
Is it intended to replace the number of items, or rather to increment it?

Reply:
replace :)